// langchain/dist/browser.cjs
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const require_runtime = require("./_virtual/_rolldown/runtime.cjs");
const require_chat_models_universal = require("./chat_models/universal.cjs");
let _langchain_core_messages = require("@langchain/core/messages");
let _langchain_core_tools = require("@langchain/core/tools");
let _langchain_core_utils_context = require("@langchain/core/utils/context");
let _langchain_core_stores = require("@langchain/core/stores");
let _langchain_core_documents = require("@langchain/core/documents");
let _langchain_core_testing = require("@langchain/core/testing");
//#region src/browser.ts
var browser_exports = /* @__PURE__ */ require_runtime.__exportAll({
	AIMessage: () => _langchain_core_messages.AIMessage,
	AIMessageChunk: () => _langchain_core_messages.AIMessageChunk,
	BaseMessage: () => _langchain_core_messages.BaseMessage,
	BaseMessageChunk: () => _langchain_core_messages.BaseMessageChunk,
	Document: () => _langchain_core_documents.Document,
	DynamicStructuredTool: () => _langchain_core_tools.DynamicStructuredTool,
	DynamicTool: () => _langchain_core_tools.DynamicTool,
	HumanMessage: () => _langchain_core_messages.HumanMessage,
	HumanMessageChunk: () => _langchain_core_messages.HumanMessageChunk,
	InMemoryStore: () => _langchain_core_stores.InMemoryStore,
	StructuredTool: () => _langchain_core_tools.StructuredTool,
	SystemMessage: () => _langchain_core_messages.SystemMessage,
	SystemMessageChunk: () => _langchain_core_messages.SystemMessageChunk,
	Tool: () => _langchain_core_tools.Tool,
	ToolMessage: () => _langchain_core_messages.ToolMessage,
	ToolMessageChunk: () => _langchain_core_messages.ToolMessageChunk,
	context: () => _langchain_core_utils_context.context,
	fakeModel: () => _langchain_core_testing.fakeModel,
	filterMessages: () => _langchain_core_messages.filterMessages,
	initChatModel: () => require_chat_models_universal.initChatModel,
	langchainMatchers: () => _langchain_core_testing.langchainMatchers,
	tool: () => _langchain_core_tools.tool,
	trimMessages: () => _langchain_core_messages.trimMessages
});
//#endregion
Object.defineProperty(exports, "AIMessage", {
	enumerable: true,
	get: function() {
		return _langchain_core_messages.AIMessage;
	}
});
Object.defineProperty(exports, "AIMessageChunk", {
	enumerable: true,
	get: function() {
		return _langchain_core_messages.AIMessageChunk;
	}
});
Object.defineProperty(exports, "BaseMessage", {
	enumerable: true,
	get: function() {
		return _langchain_core_messages.BaseMessage;
	}
});
Object.defineProperty(exports, "BaseMessageChunk", {
	enumerable: true,
	get: function() {
		return _langchain_core_messages.BaseMessageChunk;
	}
});
Object.defineProperty(exports, "Document", {
	enumerable: true,
	get: function() {
		return _langchain_core_documents.Document;
	}
});
Object.defineProperty(exports, "DynamicStructuredTool", {
	enumerable: true,
	get: function() {
		return _langchain_core_tools.DynamicStructuredTool;
	}
});
Object.defineProperty(exports, "DynamicTool", {
	enumerable: true,
	get: function() {
		return _langchain_core_tools.DynamicTool;
	}
});
Object.defineProperty(exports, "HumanMessage", {
	enumerable: true,
	get: function() {
		return _langchain_core_messages.HumanMessage;
	}
});
Object.defineProperty(exports, "HumanMessageChunk", {
	enumerable: true,
	get: function() {
		return _langchain_core_messages.HumanMessageChunk;
	}
});
Object.defineProperty(exports, "InMemoryStore", {
	enumerable: true,
	get: function() {
		return _langchain_core_stores.InMemoryStore;
	}
});
Object.defineProperty(exports, "StructuredTool", {
	enumerable: true,
	get: function() {
		return _langchain_core_tools.StructuredTool;
	}
});
Object.defineProperty(exports, "SystemMessage", {
	enumerable: true,
	get: function() {
		return _langchain_core_messages.SystemMessage;
	}
});
Object.defineProperty(exports, "SystemMessageChunk", {
	enumerable: true,
	get: function() {
		return _langchain_core_messages.SystemMessageChunk;
	}
});
Object.defineProperty(exports, "Tool", {
	enumerable: true,
	get: function() {
		return _langchain_core_tools.Tool;
	}
});
Object.defineProperty(exports, "ToolMessage", {
	enumerable: true,
	get: function() {
		return _langchain_core_messages.ToolMessage;
	}
});
Object.defineProperty(exports, "ToolMessageChunk", {
	enumerable: true,
	get: function() {
		return _langchain_core_messages.ToolMessageChunk;
	}
});
Object.defineProperty(exports, "browser_exports", {
	enumerable: true,
	get: function() {
		return browser_exports;
	}
});
Object.defineProperty(exports, "context", {
	enumerable: true,
	get: function() {
		return _langchain_core_utils_context.context;
	}
});
Object.defineProperty(exports, "fakeModel", {
	enumerable: true,
	get: function() {
		return _langchain_core_testing.fakeModel;
	}
});
Object.defineProperty(exports, "filterMessages", {
	enumerable: true,
	get: function() {
		return _langchain_core_messages.filterMessages;
	}
});
exports.initChatModel = require_chat_models_universal.initChatModel;
Object.defineProperty(exports, "langchainMatchers", {
	enumerable: true,
	get: function() {
		return _langchain_core_testing.langchainMatchers;
	}
});
Object.defineProperty(exports, "tool", {
	enumerable: true,
	get: function() {
		return _langchain_core_tools.tool;
	}
});
Object.defineProperty(exports, "trimMessages", {
	enumerable: true,
	get: function() {
		return _langchain_core_messages.trimMessages;
	}
});
