// langchain/dist/browser.js
import { __exportAll } from "./_virtual/_rolldown/runtime.js";
import { initChatModel } from "./chat_models/universal.js";
import { AIMessage, AIMessageChunk, BaseMessage, BaseMessageChunk, HumanMessage, HumanMessageChunk, SystemMessage, SystemMessageChunk, ToolMessage, ToolMessageChunk, filterMessages, trimMessages } from "@langchain/core/messages";
import { DynamicStructuredTool, DynamicTool, StructuredTool, Tool, tool } from "@langchain/core/tools";
import { context } from "@langchain/core/utils/context";
import { InMemoryStore } from "@langchain/core/stores";
import { Document } from "@langchain/core/documents";
import { fakeModel, langchainMatchers } from "@langchain/core/testing";
//#region src/browser.ts
var browser_exports = /* @__PURE__ */ __exportAll({
	AIMessage: () => AIMessage,
	AIMessageChunk: () => AIMessageChunk,
	BaseMessage: () => BaseMessage,
	BaseMessageChunk: () => BaseMessageChunk,
	Document: () => Document,
	DynamicStructuredTool: () => DynamicStructuredTool,
	DynamicTool: () => DynamicTool,
	HumanMessage: () => HumanMessage,
	HumanMessageChunk: () => HumanMessageChunk,
	InMemoryStore: () => InMemoryStore,
	StructuredTool: () => StructuredTool,
	SystemMessage: () => SystemMessage,
	SystemMessageChunk: () => SystemMessageChunk,
	Tool: () => Tool,
	ToolMessage: () => ToolMessage,
	ToolMessageChunk: () => ToolMessageChunk,
	context: () => context,
	fakeModel: () => fakeModel,
	filterMessages: () => filterMessages,
	initChatModel: () => initChatModel,
	langchainMatchers: () => langchainMatchers,
	tool: () => tool,
	trimMessages: () => trimMessages
});
//#endregion
export { AIMessage, AIMessageChunk, BaseMessage, BaseMessageChunk, Document, DynamicStructuredTool, DynamicTool, HumanMessage, HumanMessageChunk, InMemoryStore, StructuredTool, SystemMessage, SystemMessageChunk, Tool, ToolMessage, ToolMessageChunk, browser_exports, context, fakeModel, filterMessages, initChatModel, langchainMatchers, tool, trimMessages };
